src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,216 @@
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        if (inputValue === "") {
+          setOptions([]);
+          setSelectedOption(null);
+          onChange("");
+        } else {
+          const searchParams = new URLSearchParams();
+          searchParams.set("name", inputValue.toLocaleLowerCase());
+          searchParams.set("source", "zen:ticket");
+          searchParams.set("field_id", fieldId.toString());
+          searchParams.set("user_id", userId.toString());
+
+          setIsLoadingOptions(true);
+          try {
+            const response = await fetch(
+              `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+            );
+
+            const data = await response.json();
+            if (data !== undefined) {
+              setOptions(
+                data.custom_object_records.map(
+                  ({ name, id }: { name: string; id: string }) => ({
+                    name,
+                    value: id,
+                  })
+                )
+              );
+            } else {
+              setOptions([]);
+            }
+          } catch (error) {
+            return error;
+          } finally {
+            setIsLoadingOptions(false);
+          }
+        }
+      }
+      return;
+    },
+    [customObjectKey, fetchSelectedOption]
+  );
+
+  const debounceHandleChange = useMemo(
+    () => debounce(handleChange, 300),
+    [handleChange]
+  );
+
+  useEffect(() => {
+    return () => debounceHandleChange.cancel();
+  }, [debounceHandleChange]);
+
+  useEffect(() => {
+    if (value && !options.find((option) => option.value === value)) {
+      fetchSelectedOption(value as string);
+      return;
+    }
+  }, [value]);
+
+  const handleRenderValue = useCallback(() => {
+    return selectedOption?.name ?? EMPTY_OPTION.name;
+  }, [selectedOption]);
+
+  return (
+    <GardenField>
+      <Label>
+        {label}
+        {required && <Span aria-hidden="true">*</Span>}
+      </Label>
+      {description && (
+        <Hint dangerouslySetInnerHTML={{ __html: description }} />
+      )}
+      <input type="hidden" name={name} value={selectedOption?.value} />
+      <Combobox
+        inputProps={{ required }}
+        startIcon={<SearchIcon />}
+        validation={error ? "error" : undefined}
+        inputValue={inputValue}
+        selectionValue={selectedOption?.value}
+        onChange={debounceHandleChange}
+        renderValue={handleRenderValue}
+      >
+        {!required && !isLoadingOptions && (
+          <Option value="" label="-">
+            <EmptyValueOption />
+          </Option>
+        )}
+        {isLoadingOptions && (
+          <Option
+            isDisabled
+            key={loadingOption.id}
+            value={loadingOption.name}
+          />
+        )}
+        {!isLoadingOptions && !isFirstLoad && options.length === 0 && (
```

Comment:
Uhm, I am not sure I understand when we want to avoid showing "No matches found".

With the current implementation, the first time I type a letter I don't see the message but if I remove all the characters then I'll see "No matches found". Also if I select an item, then I search for a random string and I press enter, it will select the empty option. The next time I interact with the lookup field I'll see  "No matches found" as soon as I type the first character.

In general, I don't think looking for `isFirstLoad` makes much sense, but maybe I am missing something. I would rather check for `inputValue?.length > 0` so we don't show "No matches found" when the field is empty

Reply:
Ok, changed